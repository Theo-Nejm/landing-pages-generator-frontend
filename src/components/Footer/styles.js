import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    p,
    a {
      font-size: ${theme.fonts.sizes.small};
      text-align: center;
    }
  `}
`;
