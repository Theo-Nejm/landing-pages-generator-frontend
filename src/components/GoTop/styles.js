import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    color: #fff;
    background: ${theme.colors.primaryColor};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    border-radius: 50%;
    position: fixed;
    bottom: 2rem;
    right: 2rem;
    z-index: 4;
    opacity: 0.85;
  `}
`;
