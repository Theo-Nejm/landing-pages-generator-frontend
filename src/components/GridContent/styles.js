import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    text-align: center;
    max-width: 58rem;
  `}
`;

export const HTMLContainer = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xlarge} 0;
  `}
`;
