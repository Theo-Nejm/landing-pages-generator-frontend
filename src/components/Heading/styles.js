import styled, { css } from 'styled-components';

const titleSize = {
  xsmall: (theme) =>
    css`
      font-size: ${theme.fonts.sizes.xsmall};
    `,
  small: (theme) =>
    css`
      font-size: ${theme.fonts.sizes.small};
    `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,
  large: (theme) =>
    css`
      font-size: ${theme.fonts.sizes.large};
    `,
  xlarge: (theme) =>
    css`
      font-size: ${theme.fonts.sizes.xlarge};
    `,
  xxlarge: (theme) =>
    css`
      font-size: ${theme.fonts.sizes.xxlarge};
    `,
  huge: (theme) =>
    css`
      font-size: ${theme.fonts.sizes.huge};
      ${mediaFont(theme)};
    `,
  xhuge: (theme) =>
    css`
      font-size: ${theme.fonts.sizes.xhuge};
      ${mediaFont(theme)};
    `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.medium} {
    font-size: ${theme.fonts.sizes.large};
  }
`;

const titleCase = (upperCase) => css`
  text-transform: ${upperCase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, upperCase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : '#FFF'};
    ${titleSize[size](theme)};
    ${titleCase(upperCase)};
  `}
`;
