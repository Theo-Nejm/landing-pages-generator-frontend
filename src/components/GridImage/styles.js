import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    h2 {
      font-size: 4.2rem;
    }

    ${TextComponent} {
      margin-bottom: ${theme.spacings.huge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
  `}
`;

export const Element = styled.div`
  ${() => css`
    overflow: hidden;

    > img {
      width: 100%;
      transition: all 0.3s ease-in-out;

      &:hover {
        transform: scale(1.2) rotate(10deg);
      }
    }
  `}
`;
