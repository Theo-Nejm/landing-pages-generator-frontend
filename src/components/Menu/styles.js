import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';

const menuVisible = () => css`
  visibility: visible;
  opacity: 1;
`;

export const Container = styled.div`
  ${({ theme, isOpened }) => css`
    position: fixed;
    z-index: 5;
    top: 0;
    left: 0;
    right: 0;
    width: 100vw;
    transition: all .2s ease-in-out;
    height: 7rem;
    border-bottom: 1px solid ${theme.colors.mediumGray};

    > ${SectionContainer} {
      padding-top: 0;
      padding-botom: 0;
    }

    @media ${theme.media.medium} {
      height: 100vh;
      visibility: hidden;
      opacity: 0;

      ${isOpened && menuVisible}
    }
  }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: 7rem;

    @media ${theme.media.medium} {
      display: flex;
      flex-flow: column nowrap;
      text-align: center;
      align-items: center;
      justify-content: center;
      height: 100vh;

      img,
      h1 {
        margin-bottom: 2rem;
      }
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    z-index: 6;
    position: fixed;
    top: 2rem;
    right: 2rem;
    width: 3.5rem;
    height: 3.5rem;
    cursor: pointer;
    background: none;
    border: none;
    color: ${theme.colors.primaryColor};
    display: none;
    @media ${theme.media.medium} {
      display: flex;
    }
  `}
`;
