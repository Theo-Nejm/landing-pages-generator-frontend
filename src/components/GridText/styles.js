import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column nowrap;
    margin-top: 6rem;

    h2 {
      font-size: 4.2rem;
    }

    ${TextComponent} {
      margin-bottom: ${theme.spacings.huge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.medium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const Element = styled.div`
  ${({ theme }) => css`
    ${Heading} {
      position: relative;
      left: 5rem;
    }

    ${Heading}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.fonts.sizes.xhuge};
      left: -5rem;
      top: -2rem;
      transform: rotate(7.5deg);
    }
  `}
`;
