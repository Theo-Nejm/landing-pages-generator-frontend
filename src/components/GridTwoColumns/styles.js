import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};

    @media ${theme.media.medium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Title} {
      margin-bottom: ${theme.spacings.medium};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
  background: none;
  @media ${theme.media.medium} {
    margin-bottom: ${theme.spacings.medium};
  }
    }
  `}
`;

export const ImageContainer = styled.div`
  > img {
    width: 100%;
  }
`;
