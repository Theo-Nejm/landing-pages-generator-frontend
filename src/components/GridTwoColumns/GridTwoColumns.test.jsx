import { GridTwoColumns } from '.';
import { renderTheme } from '../../styles/renderTheme';
import GTCmock from './mock';

describe('<GridTwoColumns />', () => {
  it('should render <GridTwoColumns />', () => {
    const { container } = renderTheme(<GridTwoColumns {...GTCmock} />);
    expect(container).toMatchSnapshot();
  });
});
