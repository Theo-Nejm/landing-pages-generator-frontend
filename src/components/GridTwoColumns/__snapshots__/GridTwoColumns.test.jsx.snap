// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumns /> should render <GridTwoColumns /> 1`] = `
.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background: #FFF;
  color: #0a1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  color: #0a1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c6 {
  font-size: 2.4rem;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c2 .c4 {
  margin-bottom: 2.4rem;
}

.c3 {
  background: none;
}

.c7 > img {
  width: 100%;
}

@media screen and (max-width:768px) {
  .c5 {
    font-size: 3.2rem;
  }
}

@media screen and (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
    text-align: center;
  }
}

@media screen and (max-width:768px) {
  .c3 {
    margin-bottom: 2.4rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h2
            class="c4 c5"
          >
            Título da seção
          </h2>
          <p
            class="c6"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Excepturi iusto fugiat quos nobis nulla enim placeat eius nam quasi sint autem ad corporis, odit, cumque minus esse ipsam, quidem voluptate.
          </p>
        </div>
        <div
          class="c7"
        >
          <img
            alt="Título da seção"
            src="https://t.ctcdn.com.br/8FplhVkDQdAatiUcehCimgkGJlI=/512x288/smart/i257652.jpeg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
